### PR TITLE
fix #1 Invalid replace when the structure and its field are called without a getter.

### LINKED ISSUE #1: Invalid replace when the structure and its field are called without a getter.
Thanks for the project. I came across unexpected behavior when using --fix

```diff
-message Embedded {}
+message Embedded {
+  string s = 1;
+}
```

Add in test/proto.go
```diff
t.Embedded,
+t.Embedded.S,
+t.GetEmbedded().S,
)
```

Want
```diff
t.GetEmbedded(),
+t.GetEmbedded().GetS(),
+t.GetEmbedded().GetS(),
)
```

Got 
```diff
t.GetEmbedded(),
+&{t Embedded}.GetS(),
+&{%!s(*ast.SelectorExpr=&{0xc00019b120 0xc00019b140}) %!s(token.Pos=7340649) [] %!s(token.Pos=0) %!s(token.Pos=7340650)}.GetS(),
)
```

### PR DESCRIPTION
**Pull request by ghostiam**
fix #1

### CODE REVIEW

File: pkg/analyzer/proto.go [hunk 1]
```diff
@@ -69,6 +77,69 @@ func run(pass *analysis.Pass) (interface{}, error) {
...
+
+		case *ast.CallExpr:
+			v, ok := x.Fun.(*ast.SelectorExpr)
+			if ok {
+				oldExpr, newExpr, _, end2 = makeExpr(v)
+				if end2 > end {
+					end = end2
+				}
+
+				oldExpr = strings.ReplaceAll(oldExpr, "GetGet", "Get")
```
Comment (ghostiam) [inline pkg/analyzer/proto.go:126]: Not a perfect solution, but it works. Unfortunately, I don't have time to look for another solution, without copy-paste.

**Comment by ghostiam** [comment]
Due to the fact that I rewrote the entire linter :) 
I decided to add it to go-critic https://github.com/go-critic/go-critic/pull/1159

@iwittkau thanks for the sketch of the code, without it I would have developed the idea longer.

**Comment by iwittkau** [comment]
@ghostiam this is awesome. Thanks for taking over 👍🏻 

If you need any further help, let me know.

**Comment by ghostiam** [comment]
I finally completely rewrote the linter and published it, now it takes into account many different uses of fields instead of methods.
https://github.com/ghostiam/protogolint

**Pull request closed without merging**